src/openms/source/TRANSFORMATIONS/FEATUREFINDER/GaussFitter1D.cpp

```diff
@@ -89,7 +89,7 @@ namespace OpenMS
 
 
     // build model
-    model = static_cast<InterpolationModel*>(Factory<BaseModel<1> >::create("GaussModel"));
+    model = std::unique_ptr<InterpolationModel>(dynamic_cast<InterpolationModel*>(Factory<BaseModel<1>>::create("GaussModel")));
```

Comment:
maybe check if all these casts are needed.
I think the Factory could in principle directly return a std::unique_ptr<InterpolationModel>

Reply:
yes they are needed